### PR TITLE
ROE-169 Add logging interceptor

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/ROE-169

We have all the compnents:

InterceptorConfig
LoggingInterceptor
ApiLogger
Constants

Unit tests are passing

### CODE REVIEW

File: src/main/java/configuration/InterceptorConfig.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package configuration;
```
Comment (markpit) [inline src/main/java/configuration/InterceptorConfig.java:1]: package names look incorrect on a few classes

Comment (mattch1) [reply src/main/java/configuration/InterceptorConfig.java]: Changed package names and the logs are now appearing.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/Interceptor/LoggingInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.ERIC_REQUEST_ID_KEY;
+
+@Component
+public class LoggingInterceptor implements HandlerInterceptor {
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        Long startTime = System.currentTimeMillis();
+        request.getSession().setAttribute("start-time", startTime);
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/Interceptor/LoggingInterceptor.java:19]: Could define a class private constant with the "start-time" string as it appears here and also in method below.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/Interceptor/LoggingInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    public void afterCompletion(HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex) {
+        Long startTime = (Long) request.getSession().getAttribute("start-time");
+        long responseTime = System.currentTimeMillis() - startTime;
+
+        ApiLogger.infoContext(requestId(request), String.format("End of request. Method: %s Path: %s Duration: %sms Status: %s",
+                requestMethod(request), requestPath(request), responseTime, response.getStatus()), null);
+    }
+
+    private String requestPath(HttpServletRequest request) {
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/Interceptor/LoggingInterceptor.java:35]: Could prefix these three private methods with the word 'get' as they are all getting/retrieving/returning something.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/configuration/InterceptorConfig.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import uk.gov.companieshouse.overseasentitiesapi.Interceptor.LoggingInterceptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+@ComponentScan("uk.gov.companieshouse.api.interceptor")
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/configuration/InterceptorConfig.java:11]: the component scan is not needed yet, it is scanning package uk.gov.companieshouse.api.interceptor which we don't have

**Pull request merged**